zp-relayer/direct-deposit/utils.ts

```diff
@@ -0,0 +1,20 @@
+import { logger } from '@/services/appLogger'
+import { redis } from '@/services/redisClient'
+import config from '@/configs/watcherConfig'
+
+const serviceKey = 'direct-deposit'
+const lastBlockRedisKey = `${serviceKey}:lastProcessedBlock`
+
+export let lastProcessedBlock = Math.max(config.startBlock - 1, 0)
+export let lastReprocessedBlock: number
+
+export async function getLastProcessedBlock() {
```

Comment:
It seems that tracking the last processed deposit nonce would be better?

Reply:
Agree, will update

Reply:
Thought about it today, and didn't get why using last processed deposit nonce would by a good checkpoint. I mean, if watcher is restarted we still need to know last block to efficiently resume processing.

Reply:
This is about an alternative simpler approach, with fetching direct deposits through `eth_call` rather than logs, in this case we will need to store last fetched deposit nonce, and don't need to deal with blocks and events at all.

Reply:
Got it

Reply:
Btw, do you think we should bother with block confirmations in this case?

Reply:
If nothing will break on reorg which changes order of direct deposits nonces - then no

Reply:
Updated